Components/Sources/Components/Components/Watchlist/Filter/WKWatchlistFilterViewModel.swift

```diff
@@ -285,6 +275,30 @@ public final class WKWatchlistFilterViewModel {
                                         userRegistration: userRegistrationRequest,
                                         offTypes: offTypesRequest), onProjects)
     }
+
+	// MARK: - Private
+
+	private func reloadSectionData() {
+		let filterSettings = dataController.loadFilterSettings()
+		let allProjects = dataController.allWatchlistProjects()
+		let offProjects = dataController.offWatchlistProjects()
+		self.projectViewModels = Self.projectViewModels(allProjects: allProjects, offProjects: offProjects, strings: localizedStrings)
+
+		let allChangeTypes = dataController.allChangeTypes()
+		let offChangeTypes = dataController.offChangeTypes()
+
+		self.formViewModel = WKFormViewModel(sections: [
+			Self.section1(projectViewModels: Array(projectViewModels.prefix(2)), strings: localizedStrings),
+			Self.section2(projectViewModels: Array(projectViewModels.suffix(from: 2)), strings: localizedStrings, addLanguageAction: addLanguageAction),
+			Self.section3(strings: localizedStrings, filterSettings: filterSettings),
+			Self.section4(strings: localizedStrings, filterSettings: filterSettings),
+			Self.section5(strings: localizedStrings, filterSettings: filterSettings),
+			Self.section6(strings: localizedStrings, filterSettings: filterSettings),
+			Self.section7(strings: localizedStrings, filterSettings: filterSettings),
+			Self.section8(allChangeTypes: allChangeTypes, offChangeTypes: offChangeTypes, strings: localizedStrings, filterSettings: filterSettings)
+		])
+	}
```

Comment:
I think you only really need to reload section 2 here, to update it with the latest `addLanguagesAction` from that `didSet` call. You could replace this method with

```
formViewModel.sections[1] = Self.section2(projectViewModels: Array(projectViewModels.suffix(from: 2)), strings: localizedStrings, addLanguageAction: addLanguageAction)
```

and it still works.

Reply:
My intent with this approach that mimics the initializer was to capture the side effects of the user's filter settings. For example, with this suggested single line change approach, opening the filter settings, turning off a language, hitting done, and returning to the filter view still shows the language as on.